packages/jest-worker/src/index.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+const mergeStream = require('merge-stream');
+const os = require('os');
+const path = require('path');
+
+const {CHILD_MESSAGE_CALL, CHILD_MESSAGE_END} = require('./types');
+const Worker = require('./worker');
+
+import type {FarmOptions} from './types';
+import type {Readable} from 'stream';
+
+/**
+ * The Jest farm (publicly called "Worker") is a class that allows you to queue
+ * methods across multiple child processes, in order to parallelize work. This
+ * is done by providing an absolute path to a module that will be loaded on each
+ * of the child processes, and bridged to the main process.
+ *
+ * Bridged methods are specified by using the "exposedMethods" property of the
+ * options "object". This is an array of strings, where each of them corresponds
+ * to the exported name in the loaded module.
+ *
+ * You can also control the amount of workers by using the "numWorkers" property
+ * of the "options" object, and the settings passed to fork the process through
+ * the "forkOptions" property. The amount of workers defaults to the amount of
+ * CPUS minus one.
+ *
+ * Queueing calls can be done in two ways:
+ *   - Standard method: calls will be redirected to the first available worker,
+ *     so they will get executed as soon as they can.
+ *
+ *   - Sticky method: if a "computeWorkerKey" method is provided within the
+ *   config, the resulting string of this method will be used as a key.
+ *   Everytime this key is returned, it is guaranteed that your job will be
+ *   processed by the same worker. This is specially useful if your workers are
+ *   caching results.
+ */
+class Farm {
```

Comment:
Let's rename this to "Worker" like we discussed yesterday.

Reply:
I thought we agreed on using `Worker` for the farm just externally.